src/vm/i386/asmhelpers.S

```diff
@@ -850,6 +854,10 @@ NESTED_ENTRY VirtualMethodFixupStub, _TEXT, NoHandler
     // Pop ebp frame
     EPILOG_END
 
+    // Pop return address
+    add   esp, 4
+    mov   eax, ebx
```

Comment:
Now you are throwing away the address of the actual method and replacing it with the address of the thunk. This is not correct.